deploy/cr/backup-restore/restore.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: infinispan.org/v2alpha1
+kind: Restore
+metadata:
+  name: test-backup-restore
+  labels:
+    name: infinispan-operator
+spec:
+  cluster: test-backup-restore-target
+  volume:
```

Comment:
If we are going with the idea that `Backup` resource must exist in order to have something to restore from than going for `backupName: backupName` would be way simpler and all the info could be gather from there?

Reply:
I was trying to avoid making a `Backup` resource being mandatory for a restore. Imagine a user is migrating from an existing cluster that is not deployed to Openshift. They can create a `backup.zip` of the cluster via the CLI, mount a volume and transfer the .zip, then create a `Restore` resource pointing to said volume.

AFAIK the only way to add files to a Volume is to mount it to a pod and then copy the file. Do you know of a more convenient way?

Reply:
Not really. But this seems pretty inconvenient. For migration from outside of OpenShift would be the simplest for user to copy data directly to a fresh cluster and then run `bin/cli.sh -c localhost:11222 --restore <archive-file>`.

Incorporating it with Restore CR. Would require user to create some single use pod with PV, Cc data in there, create Restore CR.

Reply:
> For migration from outside of OpenShift would be the simplest for user to copy data directly to a fresh cluster and then run `bin/cli.sh -c localhost:11222 --restore <archive-file>`.

That's very true and is actually why I originally added support for uploading a backup to the server. The file-upload is netty based, but we will need to test how well this mechanism copes with large backup files of multtiple GBs

Reply:
For hybrid backup/restore I'm preferred to have a function in Infinispan console or CLI to download/upload files. Operator backup is much more for one cluster I think.

Reply:
> If we are going with the idea that `Backup` resource must exist in order to have something to restore from than going for `backupName: backupName` would be way simpler and all the info could be gather from there?

The more I think about it, the more I like it :slightly_smiling_face:. We currently have the `BackupName` field in the spec, we can just change it to `Backup` and then retrieve the backuplocation and volume spec from the Backup resource.

One thing to note, is that a Restore operation may restore a sub-set of the data contained in  a backup `.zip`. So making a Backup CR mandatory doesn't mean that `Restore.Spec.Resources`.

So the Spec would like the following:

```go
type RestoreSpec struct {
	Cluster   string                     `json:"cluster"`
	Backup    string                     `json:"backup"`
	Resources RestoreResources           `json:"resources,optional,omitempty"`
	Container v1.InfinispanContainerSpec `json:"container,optional,omitempty"`
}
```